### PR TITLE
sqlparser: Add parsing support to spatial POINT expression

### PR DESCRIPTION
**Pull request by Ayman161803**
Signed-off-by: Ayman161803 <mohammedaymannawaz@gmail.com>
This PR adds parsing support to spatial datatype expressions of the kind ``point(x,y)``

## Related Issue(s)
#8604 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

## Deployment Notes

cc @GuptaManan100

### CODE REVIEW

**Comment by Ayman161803** [comment]
I have not created new test cases because the prevailing test cases cover almost all scenarios.

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -3083,6 +3090,7 @@ func (*ExtractValueExpr) iCallable()                   {}
 func (*UpdateXMLExpr) iCallable()                      {}
 func (*PerformanceSchemaFuncExpr) iCallable()          {}
 func (*GTIDFuncExpr) iCallable()                       {}
+func (*PointExpr) iCallable()                              {}
```
Comment (dbussink) [inline go/vt/sqlparser/ast.go:3093]: I suspect the static linter checks fail on this one or the previous as they are not lined out. Please run `gofmt` for these as the CI check mentions.

File: go/vt/sqlparser/ast_format.go [hunk 2]
```diff
@@ -2754,3 +2754,10 @@ func (node *Variable) Format(buf *TrackedBuffer) {
 	}
 	buf.astPrintf(node, "%v", node.Name)
 }
+
+// Format formats the node.
+func (node *PointExpr) Format(buf *TrackedBuffer) {
+	buf.WriteString("point")
+	buf.astPrintf(node, "(%v, %v)", node.XCordinate, node.YCordinate)
+
```
Comment (dbussink) [inline go/vt/sqlparser/ast_format.go:2762]: Can you remove this stray newline?

**Comment by Ayman161803** [comment]
Addressed formatting issues

File: go/vt/sqlparser/ast_format.go [hunk 3]
```diff
@@ -2754,3 +2754,9 @@ func (node *Variable) Format(buf *TrackedBuffer) {
 	}
 	buf.astPrintf(node, "%v", node.Name)
 }
+
+// Format formats the node.
+func (node *PointExpr) Format(buf *TrackedBuffer) {
+	buf.WriteString("point")
+	buf.astPrintf(node, "(%v, %v)", node.XCordinate, node.YCordinate)
```
Comment (GuptaManan100) [inline go/vt/sqlparser/ast_format.go:2761]: Why don't you merge the first call into the second one itself like - 
```go
buf.astPrintf(node, "point (%v, %v)", node.XCordinate, node.YCordinate)
```

File: go/vt/sqlparser/sql.y [hunk 4]
```diff
@@ -7704,6 +7708,7 @@ non_reserved_keyword:
 | JSON_VALID %prec FUNCTION_CALL_NON_KEYWORD
 | JSON_VALUE %prec FUNCTION_CALL_NON_KEYWORD
 | JSON_UNQUOTE %prec FUNCTION_CALL_NON_KEYWORD
+| POINT %prec FUNCTION_CALL_NON_KEYWORD
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:7711]: Could you move this back to its original position. This list is alphabetically sorted.

**Review by GuptaManan100 — APPROVED** [review]
Everything else LGTM!

**Comment by GuptaManan100** [comment]
@Ayman161803 This is great work! Thank-you for the contribution!

**Review by frouioui — APPROVED** [review]
LGTM

File: go/vt/sqlparser/ast_format.go (hunk 3 above)
Comment (dbussink) [reply go/vt/sqlparser/ast_format.go:2761]: @GuptaManan100 How would it case it in the canonical form with this style? 

@Ayman161803 I think it's good to also add a canonical format test to the `tracked_buffer_test.go` for a `point`. So we know what it looks like / what we want it to look like there as well.

**Comment by GuptaManan100** [comment]
@Ayman161803 The file `ast_format_fast.go` isn't correctly formatted. I know it is generated but it still ends up with some formatting issues in the header.

File: go/vt/sqlparser/ast_format_fast.go [hunk 5]
```diff
@@ -19,7 +19,6 @@ package sqlparser
 
 import (
 	"fmt"
-
```
Comment (GuptaManan100) [inline go/vt/sqlparser/ast_format_fast.go:22]: This is the problem I think.

File: go/vt/sqlparser/ast_format.go (hunk 3 above)
Comment (Ayman161803) [reply go/vt/sqlparser/ast_format.go]: @dbussink I have added test cases assuming ``POINT(x, y)`` to be the right canonical form.

**Comment by Ayman161803** [comment]
@GuptaManan100 Formatted ``ast_format_fast.go`` as suggested.

@dbussink Added canonical test cases assuming ``POINT(x, y)`` to be the right canonical form.

Thank you for the reviews

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**